### ISSUE TITLE
[Monitoring V2] Ensure datasources can be dynamically reloaded without requiring Grafana redeployment

### ISSUE THREAD

**Issue by aiyengar2** [rancher/rancher#39385]
Related to https://github.com/grafana/helm-charts/issues/981

### Context

By default Rancher Monitoring allows anonymous users to be able to log in and view dashboards, since it is assumed that users who have access to the Grafana endpoint are accessing it through Rancher, which means that they are already authenticated via the Rancher API. **We do not leverage Grafana's basic authentication.**

However, as noted in the upstream issue, the admin HTTP API specifically requires basic authentication to be enabled to be able to allow datasources to be reloaded. Therefore, the behavior today in the latest Monitoring chart will be to force users to have to redeploy the Grafana workload on adding any additional datasources outside of the default one installed by the Helm chart to see any updates.

To fully support the datasource sidecar behavior of automatically picking up datasources that have been registered via ConfigMap or Secret (like we do support for dashboards, that do not require hitting an admin API to reload), we need to consider whether we'd like to enable basic authentication on Grafana by default and investigate what changes would be required to pursue this change.

**Comment by aiyengar2** [comment]
@MKlimuszka imo this is low priority since I'm not sure how many users actually run into this issue (as I'm not aware of many that add additional datasources to Grafana) and the workaround is fairly simple.

However, this should be noted in docs cc: @btat @jtravee; added release-note label to this issue

**Comment by kirannhegde** [comment]
> @MKlimuszka imo this is low priority since I'm not sure how many users actually run into this issue (as I'm not aware of many that add additional datasources to Grafana) and the workaround is fairly simple.
> 
> However, this should be noted in docs cc: @btat @jtravee; added release-note label to this issue

Is this only an issue when **additional** datasources are added to Grafana using the official kube-prometheus-stack helm chart? 
Is this not a problem when only a default datasource is defined?

 I have defined values in both the default data source as well as a value in the additional data sources . **No matter what i do, i could not get the default datasource loaded automatically using the data source side car. Even killing the grafana pod did not really help**
 The grafana datasource side car always returned the following error:
` POST request sent to http://localhost:3000/api/admin/provisioning/datasources/reload. Response: 401 Unauthorized`

I then enabled basic authentication for Grafana. However, i still continued to face the same  Unauthorized error.
 
 I then made the following changes  in the kube-prometheus-stack helm chart:
```
skipReload: true
 initDatasources: true
```
With the above changes, i now have an additional init continaer which loads the datasource at the time of creation of the grafana pod. With the above changes, i am able to get the data source updated(However, inspite of the above changes, i still have to sometimes kill the grafana pod as updation of the datasource only causes the config map to be updated and not the grafana pod to be reloaded).

So my question really is, how do we configure Grafana to reload changed data sources automatically using the kube-promehtues-stack helm chart?

Any suggestions would be very much appreciated.


Regards,
Kiran Hegde
